runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/state/WindmillStateReader.java

```diff
@@ -404,6 +415,9 @@ public void performReads() {
 
   private KeyedGetDataResponse tryGetDataFromWindmill(HashSet<StateTag<?>> stateTags)
       throws Exception {
+    if (workItemIsFailed.get()) {
+      throw new WorkItemFailedException("Windmill failed work item.");
```

Comment:
since the exception name captures it's failed, perhaps text woudl be better to indicate location
"Skipping state read as work item is no longer valid in backend"

Reply:
Done!